### DISCUSSION TITLE
Show unpublished work

### DISCUSSION THREAD

**Discussion by tomodachi94** [jj-vcs/jj#4985] [Nice Revsets] [answered]
Show work that:
* Is present locally (not necessarily bookmarked)
* Is not present on any remote

This is useful to find changes that would otherwise be forgotten.

**Comment by tomodachi94** [comment, marked as answer]
```bash
mine() ~ ::remote_bookmarks()
```

**Comment by tomodachi94** [comment]
This shows all changes not present on a remote bookmark (not necessarily mine):

```bash
~ ::remote_bookmarks()
```

This should work in theory, but in practice it has returned ancient commits (from 2016) that are definitely pushed to a remote, but that Jujutsu doesn't know are pushed to a remote. (Could this be a bug?)

**Comment by emilazy** [comment]
`mutable()` is a useful proxy for this; e.g. I use `revsets.log = "@ | ancestors(reachable(@, mutable()), 2)"` and `aliases.ll = ["log", "-r", "@ | ancestors(reachable(@ | mine(), mutable()), 2) | trunk()"]`. You do need to ensure your `immutable_heads()` is set up right for the repo.

(Though I guess you might be looking for stuff that isn’t even at the PR stage? If you delete bookmarks after their PRs are merged with the GitHub interface, they’ll disappear automatically when you fetch from your fork’s remote, so they don’t hang around too long.)

**Reply by tomodachi94** [reply]
> Though I guess you might be looking for stuff that isn’t even at the PR stage?

Yep, this was my intention, but `mutable` is interesting too (I'll need to look into customizing it) 🚀

**Comment by marco-m** [comment]
FYI this seems like mercurial `hg outgoing`: https://repo.mercurial-scm.org/hg/help/outgoing
